remoteappmanager/jupyterhub/spawners.py

```diff
@@ -25,31 +24,38 @@ class BaseSpawner(LocalProcessSpawner):
     def cmd(self):
         """Overrides the base class traitlet so that we take full control
         of the spawned command according to user admin status"""
-
         return (["remoteappadmin"]
                 if self.user.admin is True
                 else ["remoteappmanager"])
 
     def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-
-        # We get the first one. Strangely enough, jupyterhub has a table
-        # containing potentially multiple proxies, but it's enforced to
-        # contain only one.
-        self.proxy = self.db.query(orm.Proxy).first()
+        super(BaseSpawner, self).__init__(**kwargs)
+        # We can obtain a reference to the JupyterHub.proxy object
+        # through the tornado settings passed onto the User
+        self.proxy = self.user.settings.get('proxy')
 
     def get_args(self):
         args = super().get_args()
 
-        for iarg, arg in enumerate(args):
-            args[iarg] = arg.replace('--base-url=', '--base-urlpath=')
+        # Handle no default IP value (removed in jupyterhub v0.8.0)
+        if not self.ip:
+            args.append('--ip="127.0.0.1"')
```

Comment:
It's probably worth attempting to include a traitlet default here again, or if this fails document that there is a bug in `jupyterhub>0.8.0<0.9.0` that prevents doing so